### ISSUE TITLE
[Bug]: When creating a collection in Restful v2, setting isClusteringKey to true has no effect.

### ISSUE THREAD

**Issue by zhuwenxing** [milvus-io/milvus#36928]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Environment

```markdown
- Milvus version:master-aa904be-20241015
- Deployment mode(standalone or cluster):cluster
- MQ type(rocksmq, pulsar or kafka):    
- SDK version(e.g. pymilvus v2.0.0rc2):
- OS(Ubuntu or CentOS): 
- CPU/Memory: 
- GPU: 
- Others:
```

### Current Behavior

The request body has been set to true, but the corresponding attribute is not visible in the description.
```
           {
                "fieldName": "book_category",
                "dataType": "Int64",
                "isPartitionKey": false,
                "isClusteringKey": true,
                "elementTypeParams": {}
            },
```

```
 {'field_id': 103, 'name': 'book_category', 'description': '', 'type': <DataType.INT64: 5>, 'params': {}},
```

```
[2024-10-16 16:34:48 - INFO - root]: create collection test_collection_2024_10_16_16_34_48_960780uNhGoHPL with payload: {'collectionName': 'test_collection_2024_10_16_16_34_48_960780uNhGoHPL', 'params': {'shardsNum': '4', 'partitionsNum': '16', 'consistencyLevel': 'Strong', 'ttlSeconds': '60'}, 'schema': {'enableDynamicField': True, 'fields': [{'fieldName': 'user_id', 'dataType': 'Int64', 'isPrimary': False, 'elementTypeParams': {}}, {'fieldName': 'book_id', 'dataType': 'Int64', 'isPrimary': True, 'elementTypeParams': {}}, {'fieldName': 'word_count', 'dataType': 'Int64', 'isPartitionKey': True, 'isClusteringKey': False, 'elementTypeParams': {}}, {'fieldName': 'book_category', 'dataType': 'Int64', 'isPartitionKey': False, 'isClusteringKey': True, 'elementTypeParams': {}}, {'fieldName': 'book_describe', 'dataType': 'VarChar', 'elementTypeParams': {'max_length': '256'}}, {'fieldName': 'json', 'dataType': 'JSON', 'elementTypeParams': {}}, {'fieldName': 'int_array', 'dataType': 'Array', 'elementDataType': 'Int64', 'elementTypeParams': {'max_capacity': '1024'}}, {'fieldName': 'book_intro', 'dataType': 'FloatVector', 'elementTypeParams': {'dim': '128'}}]}, 'indexParams': [{'fieldName': 'book_intro', 'indexName': 'book_intro_vector', 'metricType': 'L2', 'indexType': 'AUTOINDEX', 'params': {}}]} (test_collection_operations.py:221)
[2024-10-16 16:34:49 - DEBUG - urllib3.connectionpool]: Starting new HTTP connection (1): 10.104.18.242:19530 (connectionpool.py:246)
[2024-10-16 16:34:53 - DEBUG - urllib3.connectionpool]: http://10.104.18.242:19530 "POST /v2/vectordb/collections/create HTTP/1.1" 200 20 (connectionpool.py:475)
[2024-10-16 16:34:53 - DEBUG - ci_test]: 
method: post, 
url: http://10.104.18.242:19530/v2/vectordb/collections/create, 
cost time: 3.788130044937134, 
header: {'Content-Type': 'application/json', 'Authorization': 'Bearer None', 'RequestId': '815d8e40-8b99-11ef-b1b2-acde48001122'}, 
payload: {
    "collectionName": "test_collection_2024_10_16_16_34_48_960780uNhGoHPL",
    "params": {
        "shardsNum": "4",
        "partitionsNum": "16",
        "consistencyLevel": "Strong",
        "ttlSeconds": "60"
    },
    "schema": {
        "enableDynamicField": true,
        "fields": [
            {
                "fieldName": "user_id",
                "dataType": "Int64",
                "isPrimary": false,
                "elementTypeParams": {}
            },
            {
                "fieldName": "book_id",
                "dataType": "Int64",
                "isPrimary": true,
                "elementTypeParams": {}
            },
            {
                "fieldName": "word_count",
                "dataType": "Int64",
                "isPartitionKey": true,
                "isClusteringKey": false,
                "elementTypeParams": {}
            },
            {
                "fieldName": "book_category",
                "dataType": "Int64",
                "isPartitionKey": false,
                "isClusteringKey": true,
                "elementTypeParams": {}
            },
            {
                "fieldName": "book_describe",
                "dataType": "VarChar",
                "elementTypeParams": {
                    "max_length": "256"
                }
            },
            {
                "fieldName": "json",
                "dataType": "JSON",
                "elementTypeParams": {}
            },
            {
                "fieldName": "int_array",
                "dataType": "Array",
                "elementDataType": "Int64",
                "elementTypeParams": {
                    "max_capacity": "1024"
                }
            },
            {
                "fieldName": "book_intro",
                "dataType": "FloatVector",
                "elementTypeParams": {
                    "dim": "128"
                }
            }
        ]
    },
    "indexParams": [
        {
            "fieldName": "book_intro",
            "indexName": "book_intro_vector",
            "metricType": "L2",
            "indexType": "AUTOINDEX",
            "params": {}
        }
    ]
}, 
response: {"code":0,"data":{}} (milvus.py:77)
[2024-10-16 16:34:53 - DEBUG - urllib3.connectionpool]: Starting new HTTP connection (1): 10.104.18.242:19530 (connectionpool.py:246)
[2024-10-16 16:34:53 - DEBUG - urllib3.connectionpool]: http://10.104.18.242:19530 "POST /v2/vectordb/collections/list HTTP/1.1" 200 72 (connectionpool.py:475)
[2024-10-16 16:34:53 - DEBUG - ci_test]: 
method: post, 
url: http://10.104.18.242:19530/v2/vectordb/collections/list, 
cost time: 0.054605960845947266, 
header: {'Content-Type': 'application/json', 'Authorization': 'Bearer None', 'RequestId': '815d8e40-8b99-11ef-b1b2-acde48001122'}, 
payload: {}, 
response: {"code":0,"data":["test_collection_2024_10_16_16_34_48_960780uNhGoHPL"]} (milvus.py:77)
[2024-10-16 16:34:53 - INFO - ci_test]: pymilvus describe collection: {'collection_name': 'test_collection_2024_10_16_16_34_48_960780uNhGoHPL', 'auto_id': False, 'num_shards': 4, 'description': '', 'fields': [{'field_id': 100, 'name': 'user_id', 'description': '', 'type': <DataType.INT64: 5>, 'params': {}}, {'field_id': 101, 'name': 'book_id', 'description': '', 'type': <DataType.INT64: 5>, 'params': {}, 'is_primary': True}, {'field_id': 102, 'name': 'word_count', 'description': '', 'type': <DataType.INT64: 5>, 'params': {}, 'is_partition_key': True}, {'field_id': 103, 'name': 'book_category', 'description': '', 'type': <DataType.INT64: 5>, 'params': {}}, {'field_id': 104, 'name': 'book_describe', 'description': '', 'type': <DataType.VARCHAR: 21>, 'params': {'max_length': 256}}, {'field_id': 105, 'name': 'json', 'description': '', 'type': <DataType.JSON: 23>, 'params': {}}, {'field_id': 106, 'name': 'int_array', 'description': '', 'type': <DataType.ARRAY: 22>, 'params': {'max_capacity': 1024}, 'element_type': <DataType.INT64: 5>}, {'field_id': 107, 'name': 'book_intro', 'description': '', 'type': <DataType.FLOAT_VECTOR: 101>, 'params': {'dim': 128}}], 'aliases': [], 'collection_id': 453264603869874368, 'consistency_level': 0, 'properties': {'collection.ttl.seconds': '60'}, 'num_partitions': 16, 'enable_dynamic_field': True} (test_collection_operations.py:231)
[2024-10-16 16:35:03 - DEBUG - urllib3.connectionpool]: Starting new HTTP connection (1): 10.104.18.242:19530 (connectionpool.py:246)
[2024-10-16 16:35:03 - DEBUG - urllib3.connectionpool]: http://10.104.18.242:19530 "POST /v2/vectordb/collections/describe HTTP/1.1" 200 1569 (connectionpool.py:475)
[2024-10-16 16:35:03 - DEBUG - ci_test]: 
method: post, 
url: http://10.104.18.242:19530/v2/vectordb/collections/describe, 
cost time: 0.038642168045043945, 
header: {'Content-Type': 'application/json', 'Authorization': 'Bearer None', 'RequestId': '815d8e40-8b99-11ef-b1b2-acde48001122'}, 
payload: {
    "collectionName": "test_collection_2024_10_16_16_34_48_960780uNhGoHPL"
}, 
response: {"code":0,"data":{"aliases":[],"autoId":false,"collectionID":453264603869874368,"collectionName":"test_collection_2024_10_16_16_34_48_960780uNhGoHPL","consistencyLevel":"Strong","description":"","enab...":"book_intro","indexName":"book_intro_vector","metricType":"L2"}],"load":"LoadStateLoaded","partitionsNum":16,"properties":[{"key":"collection.ttl.seconds","value":"60"}],"shardsNum":4},"message":""} (milvus.py:77)
[2024-10-16 16:35:03 - INFO - ci_test]: restful describe collection: {'code': 0, 'data': {'aliases': [], 'autoId': False, 'collectionID': 453264603869874368, 'collectionName': 'test_collection_2024_10_16_16_34_48_960780uNhGoHPL', 'consistencyLevel': 'Strong', 'description': '', 'enableDynamicField': True, 'fields': [{'autoId': False, 'description': '', 'id': 100, 'name': 'user_id', 'partitionKey': False, 'primaryKey': False, 'type': 'Int64'}, {'autoId': False, 'description': '', 'id': 101, 'name': 'book_id', 'partitionKey': False, 'primaryKey': True, 'type': 'Int64'}, {'autoId': False, 'description': '', 'id': 102, 'name': 'word_count', 'partitionKey': True, 'primaryKey': False, 'type': 'Int64'}, {'autoId': False, 'description': '', 'id': 103, 'name': 'book_category', 'partitionKey': False, 'primaryKey': False, 'type': 'Int64'}, {'autoId': False, 'description': '', 'id': 104, 'name': 'book_describe', 'params': [{'key': 'max_length', 'value': '256'}], 'partitionKey': False, 'primaryKey': False, 'type': 'VarChar'}, {'autoId': False, 'description': '', 'id': 105, 'name': 'json', 'partitionKey': False, 'primaryKey': False, 'type': 'JSON'}, {'autoId': False, 'description': '', 'elementType': 'Int64', 'id': 106, 'name': 'int_array', 'params': [{'key': 'max_capacity', 'value': '1024'}], 'partitionKey': False, 'primaryKey': False, 'type': 'Array'}, {'autoId': False, 'description': '', 'id': 107, 'name': 'book_intro', 'params': [{'key': 'dim', 'value': '128'}], 'partitionKey': False, 'primaryKey': False, 'type': 'FloatVector'}], 'functions': None, 'indexes': [{'fieldName': 'book_intro', 'indexName': 'book_intro_vector', 'metricType': 'L2'}], 'load': 'LoadStateLoaded', 'partitionsNum': 16, 'properties': [{'key': 'collection.ttl.seconds', 'value': '60'}], 'shardsNum': 4}, 'message': ''} (test_collection_operations.py:235)
rsp
PyDev console: starting.
{'code': 0, 'data': {'aliases': [], 'autoId': False, 'collectionID': 453264603869874368, 'collectionName': 'test_collection_2024_10_16_16_34_48_960780uNhGoHPL', 'consistencyLevel': 'Strong', 'description': '', 'enableDynamicField': True, 'fields': [{'autoId': False, 'description': '', 'id': 100, 'name': 'user_id', 'partitionKey': False, 'primaryKey': False, 'type': 'Int64'}, {'autoId': False, 'description': '', 'id': 101, 'name': 'book_id', 'partitionKey': False, 'primaryKey': True, 'type': 'Int64'}, {'autoId': False, 'description': '', 'id': 102, 'name': 'word_count', 'partitionKey': True, 'primaryKey': False, 'type': 'Int64'}, {'autoId': False, 'description': '', 'id': 103, 'name': 'book_category', 'partitionKey': False, 'primaryKey': False, 'type': 'Int64'}, {'autoId': False, 'description': '', 'id': 104, 'name': 'book_describe', 'params': [{'key': 'max_length', 'value': '256'}], 'partitionKey': False, 'primaryKey': False, 'type': 'VarChar'}, {'autoId': False, 'description': '', 'id': 105, 'name': 'json', 'partitionKey': False, 'primaryKey': False, 'type': 'JSON'}, {'autoId': False, 'description': '', 'elementType': 'Int64', 'id': 106, 'name': 'int_array', 'params': [{'key': 'max_capacity', 'value': '1024'}], 'partitionKey': False, 'primaryKey': False, 'type': 'Array'}, {'autoId': False, 'description': '', 'id': 107, 'name': 'book_intro', 'params': [{'key': 'dim', 'value': '128'}], 'partitionKey': False, 'primaryKey': False, 'type': 'FloatVector'}], 'functions': None, 'indexes': [{'fieldName': 'book_intro', 'indexName': 'book_intro_vector', 'metricType': 'L2'}], 'load': 'LoadStateLoaded', 'partitionsNum': 16, 'properties': [{'key': 'collection.ttl.seconds', 'value': '60'}], 'shardsNum': 4}, 'message': ''}
res = c.describe()
res
{'collection_name': 'test_collection_2024_10_16_16_34_48_960780uNhGoHPL', 'auto_id': False, 'num_shards': 4, 'description': '', 'fields': [{'field_id': 100, 'name': 'user_id', 'description': '', 'type': <DataType.INT64: 5>, 'params': {}}, {'field_id': 101, 'name': 'book_id', 'description': '', 'type': <DataType.INT64: 5>, 'params': {}, 'is_primary': True}, {'field_id': 102, 'name': 'word_count', 'description': '', 'type': <DataType.INT64: 5>, 'params': {}, 'is_partition_key': True}, {'field_id': 103, 'name': 'book_category', 'description': '', 'type': <DataType.INT64: 5>, 'params': {}}, {'field_id': 104, 'name': 'book_describe', 'description': '', 'type': <DataType.VARCHAR: 21>, 'params': {'max_length': 256}}, {'field_id': 105, 'name': 'json', 'description': '', 'type': <DataType.JSON: 23>, 'params': {}}, {'field_id': 106, 'name': 'int_array', 'description': '', 'type': <DataType.ARRAY: 22>, 'params': {'max_capacity': 1024}, 'element_type': <DataType.INT64: 5>}, {'field_id': 107, 'name': 'book_intro', 'description': '', 'type': <DataType.FLOAT_VECTOR: 101>, 'params': {'dim': 128}}], 'aliases': [], 'collection_id': 453264603869874368, 'consistency_level': 0, 'properties': {'collection.ttl.seconds': '60'}, 'num_partitions': 16, 'enable_dynamic_field': True}

```

**Comment by zhuwenxing** [comment]
/assign @smellthemoon 
PTAL

**Comment by zhuwenxing** [comment]
when using pymilvus to do the same thing, cluster key can be set success
```
from pymilvus import DataType, FieldSchema, CollectionSchema, Collection, connections

connections.connect(host="10.104.18.242")
dim=128


default_fields = [
    FieldSchema(name="id", dtype=DataType.INT64, is_primary=True),
    FieldSchema(name="key", dtype=DataType.INT64, is_clustering_key=True),
    FieldSchema(name="var", dtype=DataType.VARCHAR, max_length=1000, is_primary=False),
    FieldSchema(name="embeddings", dtype=DataType.FLOAT_VECTOR, dim=dim)
]

default_schema = CollectionSchema(fields=default_fields, description="test clustering-key collection")
coll1 = Collection(name="clustering_key_scalar", schema=default_schema)
res = coll1.describe()
print(res)
```
output
`{'field_id': 101, 'name': 'key', 'description': '', 'type': <DataType.INT64: 5>, 'params': {}, 'is_clustering_key': True}, `
```
{'collection_name': 'clustering_key_scalar', 'auto_id': False, 'num_shards': 1, 'description': 'test clustering-key collection', 'fields': [{'field_id': 100, 'name': 'id', 'description': '', 'type': <DataType.INT64: 5>, 'params': {}, 'is_primary': True}, {'field_id': 101, 'name': 'key', 'description': '', 'type': <DataType.INT64: 5>, 'params': {}, 'is_clustering_key': True}, {'field_id': 102, 'name': 'var', 'description': '', 'type': <DataType.VARCHAR: 21>, 'params': {'max_length': 1000}}, {'field_id': 103, 'name': 'embeddings', 'description': '', 'type': <DataType.FLOAT_VECTOR: 101>, 'params': {'dim': 128}}], 'aliases': [], 'collection_id': 453264603869876702, 'consistency_level': 2, 'properties': {}, 'num_partitions': 1, 'enable_dynamic_field': False}
```

**Comment by zhuwenxing** [comment]
At that time, after setting the is_clustering_key to true using pymilvus, the describe interface of pymilvus can see that the is_clustering_key setting is true. However, if the collection is described using the restful v2 interface, the is_clustering_key will not be displayed.

**Comment by smellthemoon** [comment]
pr merged. could you please help to verify? @zhuwenxing

**Comment by zhuwenxing** [comment]
/assign

verified and fixed in `master-2e76709-20241019`
